### ISSUE TITLE
Warn about "useless blocks"

### ISSUE THREAD

**Issue by MatrixFrog** [google/closure-compiler#1458]
Inspired by some conversations with @concavelenz and tweets by @cramforce, we should warn about an unnecessary use of a block. A block is considered useful if it is part of an if / else / while / switch / etc. or if it contains any block-scoped variables (let, const, class, or function declarations). Otherwise there is no reason to use it so it is likely a mistake. This would catch the classic error

```javascript
return
    {foo: 'bar'};
```
as well as the contrived but plausible "backdoor" example at https://twitter.com/cramforce/status/692197974957527040

```javascript
if (denied) {
  showAccessDenied()
} {
  grantAccess()
}
```

It might also catch some cases where a user expected a `var`-declared variable to go out of scope because they mistakenly believed that `var`-declared variables are block-scoped.
